src/cache/lru_cache_manager/lru_node.h

```diff
@@ -29,6 +29,6 @@ struct LRUNode {
  * @param  resultSet: Node's result value (for CacheData)
  * @retval Initilized LRU node (pointer)
  */
-LRUNode *initLRUNode(LRUNode* node, const char *hashKey, ResultSet* resultSet);
+LRUNode *initLRUNode(LRUNode *node, unsigned long long const hashKey, void *cacheValue, cacheValueFreeFunc freeCB);
```

Comment:
Assuming all entries in the LRU queue are of the same type, freeCB should be a member of the cache object.
Also I would assume the LRU queue to allocate a single consecutive array of LRUNode objects.

Reply:
the actual free is called from the queue